src/wizards/tokenSwapDealWizard/wizardManager.ts

```diff
@@ -161,7 +165,7 @@ export class WizardManager {
     }
 
     await deal.ensureInitialized();
-    return deal.registrationData;
+    return JSON.parse(JSON.stringify(deal.registrationData));
```

Comment:
I suspect that  `Object.assign({}, deal.registrationData)` would be more efficient.

Reply:
I've testing it and Object.assign doesn't deep clone. I've checked on MDN https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#warning_for_deep_clone